classes/Tab.php

```diff
@@ -682,15 +682,44 @@ public static function getClassNameById($idTab)
         return Db::getInstance()->getValue('SELECT class_name FROM ' . _DB_PREFIX_ . 'tab WHERE id_tab = ' . (int) $idTab);
     }
 
+    /**
+     * @param string $file_to_refresh
+     * @param string $external_file
+     *
+     * @return bool
+     */
+    private static function refresh($file_to_refresh, $external_file)
+    {
+        $content = Tools::file_get_contents($external_file);
+
+        return $content ? (bool) file_put_contents(_PS_ROOT_DIR_ . $file_to_refresh, $content) : false;
```

Comment:
```suggestion
        return $content ? (bool) file_put_contents(_PS_ROOT_DIR_ . $file_to_refresh, $content) : false;
```
file_put_contents should not be cast in bool, inserting 0 bytes will return false whereas everything is ok :thinking: 

Otherwise you can use `Tools::copy` which is doing the same thing